app/services/user_signup.rb

```diff
@@ -0,0 +1,49 @@
+class UserSignup
+  attr_reader :error_message
+
+  def initialize(user)
+    @user = user
+  end
+
+  def sign_up(stripe_token, invitation_token)
+    if @user.valid?
+      charge = StripeWrapper::Charge.create(
+        amount: 999,
+        card: stripe_token,
+        description: "Sign up charge for #{@user.email}"
+      )
+      if charge.successful?
+        @user.save
+        handle_invitation(invitation_token)
+        AppMailer.send_welcome_email(@user).deliver
+        @status = :success
+        self
+      else
+        @status = :failed
+        @error_message = charge.error_message
+        self
+      end
+    else
+      @status = :failed
+      @error_message = "Invalid user info. Please check the below errors and try again."
+      self
```

Comment:
You could refactor all `self`s to the end of this method. All `if` branches return it.